docs/package-deletion-policies.md

```diff
@@ -0,0 +1,210 @@
+
+## Background
+
+In 2016 after a dispute about the naming of an open source package, a maintainer deleted all the packages he had published to npm, including a small package called `left-pad`.
+Although `left-pad` was only about a dozen lines of code, it had impressive reach.
+Prior to removal, it had over 15 million downloads and was used by thousands of open source projects--including many that are critical to the npm ecosystem.
+The deletion of `left-pad` caused build failures for projects large and small and rendered a large portion of the internet unnavigable. Npm made the decision to republish the package.
+
+[After the `left-pad` incident](https://en.wikipedia.org/wiki/Npm_left-pad_incident), npm updated their package deletion policy to prevent similar failures and other package managers followed suit.
+In the over 9 years since the incident, package managers have continued to adjust their package deletion policies to balance the sometimes competing interests of open source maintainers, open source consumers, and the package ecosystem itself.
+
+This paper outlines the components of modern package deletion policies.
+It is not meant to be a sample deletion policy, but a list of things to consider when crafting such a policy including package deletion alternatives.
+
+## Deletion Policy Considerations
+
+If we consider a hypothetical package deletion policy, it would exist on a spectrum somewhere between never allowing packages to be deleted to allowing packages to be deleted without any restrictions.
+Both ends of this spectrum are impractical.
+
+* Never allowing for package deletion could result in the publishing of malicious, abusive, illegal, or copywritten content without any recourse.
+* Allowing for unrestricted package deletion could result in build failures like those seen in the `left-pad` incident, malicious packages snapping up a recently vacated namespace, and other undesirable behavior.
+
+So when crafting a deletion policy, we want to stay away from the two extremes, but what needs to be considered?
+
+Whether or not a package manager allows for users to delete packages in some cases, it should allow for the package manager maintainers to manually intervene and delete packages.
+This allows the maintainer team to protect the ecosystem from malicious, abusive, illegal, or copyrighted content.
+
+But what if a user wants to delete a package they published?
+Although it's possible to require such cases to go to the maintainer team to be manually reviewed and removed, that is not a very efficient policy.
+Such a policy would place a burden on the (already likely overworked) maintainer team, and the ecosystem would be cluttered with the unwanted packages until they are manually removed.
+
+Allowing a user to delete their package in certain cases is therefore good for the user, the package manager maintainer team, and the package ecosystem at large.
+But what factors need to be considered to avoid a `left-pad` style incident?
```

Comment:
... while still avoiding illegal or unsafe content.

I think you have to hold both in each hand, pretty much all the time. For example if we had to left-pad the world because of some DMCA'd or malicious content, I think you'd have to as the repo admin.

NuGet.org has deleted relatively popular content that we found was infringing copyright.